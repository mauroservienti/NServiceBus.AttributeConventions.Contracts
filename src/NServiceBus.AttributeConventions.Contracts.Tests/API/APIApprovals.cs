using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using PublicApiGenerator;
using System.Runtime.CompilerServices;

namespace NServiceBus.AttributeConventions.Contracts.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(MessageAttribute).Assembly, options: null);

            Approvals.Verify(publicApi);
        }
    }
}
