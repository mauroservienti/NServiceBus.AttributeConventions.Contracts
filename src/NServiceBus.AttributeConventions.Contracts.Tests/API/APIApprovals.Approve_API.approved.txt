[assembly: System.Reflection.AssemblyMetadata("RepositoryUrl", "")]
[assembly: System.Runtime.Versioning.TargetFramework(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace NServiceBus.AttributeConventions.Contracts
{
    [System.AttributeUsage(System.AttributeTargets.Class | System.AttributeTargets.All, AllowMultiple=false, Inherited=false)]
    public sealed class CommandAttribute : System.Attribute
    {
        public CommandAttribute() { }
    }
    [System.AttributeUsage(System.AttributeTargets.Class | System.AttributeTargets.Interface | System.AttributeTargets.All, AllowMultiple=false, Inherited=false)]
    public sealed class EventAttribute : System.Attribute
    {
        public EventAttribute() { }
    }
    [System.AttributeUsage(System.AttributeTargets.Class | System.AttributeTargets.All, AllowMultiple=false, Inherited=false)]
    public sealed class MessageAttribute : System.Attribute
    {
        public MessageAttribute() { }
    }
}